lifecycle/autodispose-lifecycle-ktx/src/test/kotlin/com/uber/autodispose/lifecycle/ktx/AutoDisposeLifecycleTest.kt

```diff
@@ -0,0 +1,318 @@
+
+    o.assertValue { it == "Hello" }
+    o.assertComplete()
+  }
+
+  @Test fun observable_lifecycleNormalInterrupted() {
+    lifecycleScopeProvider.start()
+    val subject = PublishSubject.create<String>()
+    subject
+        .autoDisposable(lifecycleScopeProvider)
+        .subscribe(o)
+
+    subject.onNext("Hello")
+
+    o.assertValue { it == "Hello" }
+
+    lifecycleScopeProvider.stop()
+
+    // https://github.com/ReactiveX/RxJava/issues/5178
+//    assertThat(o.isDisposed).isTrue()
+//    o.assertNotSubscribed()
+  }
+
+  @Test fun observable_lifecycleEnded() {
+    lifecycleScopeProvider.start()
+    lifecycleScopeProvider.stop()
+    Observable.just("Hello")
+        .autoDisposable(lifecycleScopeProvider)
+        .subscribe(o)
+
+    o.assertError { it is com.uber.autodispose.lifecycle.LifecycleEndedException }
+  }
+
+  @Test fun flowable_lifecycleNotStarted() {
+    Flowable.just("Hello")
+        .autoDisposable(lifecycleScopeProvider)
+        .subscribe(s)
+
+    s.assertError { it is com.uber.autodispose.lifecycle.LifecycleNotStartedException }
+  }
+
+  @Test fun flowable_lifecycleNormalCompletion() {
+    lifecycleScopeProvider.start()
+    Flowable.just("Hello")
+        .autoDisposable(lifecycleScopeProvider)
+        .subscribe(s)
+
+    s.assertValue { it == "Hello" }
+    s.assertComplete()
+  }
+
+  @Test fun flowable_lifecycleNormalInterrupted() {
+    lifecycleScopeProvider.start()
+    val subject = PublishSubject.create<String>()
+    subject.toFlowable(ERROR)
+        .autoDisposable(lifecycleScopeProvider)
+        .subscribe(s)
+
+    subject.onNext("Hello")
+
+    s.assertValue { it == "Hello" }
+
+    lifecycleScopeProvider.stop()
+
+    // https://github.com/ReactiveX/RxJava/issues/5178
+//    assertThat(s.isDisposed).isTrue()
+//    s.assertNotSubscribed()
+  }
+
+  @Test fun flowable_lifecycleEnded() {
+    lifecycleScopeProvider.start()
+    lifecycleScopeProvider.stop()
+    Flowable.just("Hello")
+        .autoDisposable(lifecycleScopeProvider)
+        .subscribe(s)
+
+    s.assertError { it is com.uber.autodispose.lifecycle.LifecycleEndedException }
+  }
+
+  @Test fun maybe_lifecycleNotStarted() {
+    Maybe.just("Hello")
+        .autoDisposable(lifecycleScopeProvider)
+        .subscribe(o)
+
+    o.assertError { it is com.uber.autodispose.lifecycle.LifecycleNotStartedException }
+  }
+
+  @Test fun maybe_lifecycleNormalCompletion() {
+    lifecycleScopeProvider.start()
+    Maybe.just("Hello")
+        .autoDisposable(lifecycleScopeProvider)
+        .subscribe(o)
+
+    o.assertValue { it == "Hello" }
+    o.assertComplete()
+  }
+
+  @Test fun maybe_lifecycleNormalInterrupted() {
+    lifecycleScopeProvider.start()
+    val subject = PublishSubject.create<String>()
+    subject
+        .autoDisposable(lifecycleScopeProvider)
+        .subscribe(o)
+
+    lifecycleScopeProvider.stop()
+
+    // https://github.com/ReactiveX/RxJava/issues/5178
+//    assertThat(o.isDisposed).isTrue()
```

Comment:
kill?

Reply:
We leave these to indicate what we _would_ assert/expect there but can't due to the linked issue